src/Common/macros.jl

```diff
@@ -0,0 +1,83 @@
+# ----------------------------------------------------------------------------------------------------------
+#                                                CALLBACK 
+@doc raw"""
+    @Common.cbcall callbacks::Tuple{CallbackObject, N} step::Int64 Vararg::Any
+
+(Macro) Call the CallbackObject.function of each [`CallbackObject`](@ref) in the `callbacks` Tuple{CallbackObject, N} independently.
+Each CallbackObject.function is ran depending on the defined CallbackObject.freq and the given `step`.
+Vararg holds all the arguments necessary to run the callback function itself.
+
+# Examples
+```julia-repl
+julia> @Common.cbcall (callback_object1, callback_object2) 1
+```
+"""
+macro cbcall(cbs, step, args...)
+
+    ex = quote
+        for cb in $cbs
+            if (getproperty(cb, :freq)>0) && ($step%getproperty(cb, :freq)==0)
+                getproperty(cb, :callback)($step, $(args...))
+            end
+        end
+    end
+    return esc(ex)
+end
+
+
+@doc raw"""
+    @Common.callback f::Function freq::Int64
+
+(Macro) Create a [`CallbackObject`](@ref) with the given function `f` and output frequency `freq`. 
+
+# Examples
+```julia-repl
+julia> @Common.callback my_callback1 10
+CallbackObject(callback=my_callback1, freq=10)
+```
+"""
+macro callback(f, freq)
```

Comment:
```suggestion
macro callback(args...)
    ex = :(Common.CallbackObject($(args...)))
    esc(ex)
end
```
assim podemos definir o `CallbackObject` com número variável de argumentos, sendo que se apenas passarmos a função, então `freq=1`